File: src/models/item.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

// Item used on the packages.
// They can be publicity items (posts on Facebook), number of stand days, size of logo on the website, etc
type Item struct {

	// Item's ID (_id of mongodb).
	ID primitive.ObjectID `json:"id" bson:"_id"`

	Name string `json:"name" bson:"name"`

	// Type represents the type of item. This is dynamic and decided on each Event by the Coordination,
	// but it can be, for example, "Publicity", "Merchandise", "Stands", "Talk"...
	Type        string `json:"type" bson:"type"`
	Description string `json:"description" bson:"description"`

	// This is a visual representation of the item. For example, in a merchandise item, it could be a photo.
	Image string `json:"img" bson:"img"`

	// Price in cents (€).
	Price int `json:"price" bson:"price"`

	// Tax percentage. The total value will be Price * (1 + VAT).
	VAT int `json:"vat" bson:"vat"`
}

File: frontend/lib/models/item.dart
import 'dart:convert';

class Item {
  final String id;
  final String name;
  final String type;
  final String description;
  final String image;
  final int price;
  final int vat;

  Item({required this.id, required this.name, required this.type, required this.description, required this.image, required this.price, required this.vat});

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: json['id'],
      name: json['name'],
      type: json['type'],
      description: json['description'],
      image: json['img'],
      price: json['price'],
      vat: json['vat'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'type': type,
        'description': description,
        'img': image,
        'price': price,
        'vat': vat,
  };

  @override
  String toString() {
    return json.encode(this.toJson());
  }
}
